internal/internal_worker_base.go

```diff
@@ -260,6 +276,14 @@ func (bw *baseWorker) pollTask() {
 	var task interface{}
 	bw.retrier.Throttle()
 	if bw.pollLimiter == nil || bw.pollLimiter.Wait(bw.limiterContext) == nil {
+		if bw.pollerAutoScaler != nil {
+			if pErr := bw.pollerAutoScaler.Acquire(1); pErr == nil {
+				defer bw.pollerAutoScaler.Release(1)
+			} else {
+				bw.logger.Warn("poller auto scaler acquire error", zap.Error(pErr))
+			}
+		}
```

Comment:
I'm pretty sure this ordering means this is possible:
- autoscale from N to 1
- allow N to get through pollLimiter, while autoscaler allows one long request and blocks N-1
- long request finishes, N-1 run through as fast as possible, ignoring ratelimter
  - sequentially, as autoscaler is at 1.  or 2 at once if at 2, etc.

swapping the order seems better, since "waiting on the per-second request limiter" feels like "a concurrent poller" to me.  the request itself isn't running, but we've reserved space for it to run "on time".  so we'll degrade "concurrent requests on the network" (allowing it to go lower than specified) in favor of preserving "requests per second".